### PR TITLE
Update re-templating grid filter 2621

### LINKED ISSUE #2621: Re-templating filter cell samples are loaded with no data
### Description  
The samples demonstrating how the filter cell can be retemplated are not loading their data.

### Steps to reproduce  

1. Navigate to [Re-templating Filter Cell ](https://staging.infragistics.com/products/ignite-ui-angular/angular/components/treegrid/filtering#re-templating-filter-cell)
2. Check the demo

### Result  
The grid is empty.

### Expected result  
The grid should show the data it's bound to.

Note: It happens in the IgxGrid and IgxTreeGrid samples where we have date columns. The igxDatePickerTemplate has been removed so we need to update the samples accordingly.

### PR DESCRIPTION
**Pull request by zhosti**
Changing date filter in all grid samples for re-templating filter cell 

closes #2621

### CODE REVIEW

File: src/app/grid/grid-filtering-template-sample/grid-filtering-template-sample.component.html [hunk 1]
```diff
@@ -42,31 +42,11 @@
     </ng-template>
     <ng-template #dateFilterTemplate igxFilterCellTemplate let-column="column">
         <div class="filter-cell">
-            <igx-date-picker [value]="getFilterValue(column)"
-                             (valueChange)="onDateSelected($event, column)" mode="dialog">
+            <igx-date-picker [value]="getFilterValue(column)" (valueChange)="onDateSelected($event, column)" mode="dialog">
```
Comment (gedinakova) [inline src/app/grid/grid-filtering-template-sample/grid-filtering-template-sample.component.html:45]: In order to achieve the same appearance as the other three filter cells, we need to change the date picker template a bit more:
```
<igx-date-picker #dp [value]="getFilterValue(column)" (valueChange)="onDateSelected($event, column)" mode="dialog"
                [displayDensity]="'compact'" placeholder="Filter..." type="box">
                <igx-picker-toggle (click)="dp.open()" igxPrefix>
                    <igx-icon>search</igx-icon>
                </igx-picker-toggle>
                <igx-picker-clear *ngIf="value" igxSuffix (click)="dp.clear()">
                    <igx-icon>clear</igx-icon>
                </igx-picker-clear>
            </igx-date-picker>
```

File: src/app/grid/grid-filtering-template-sample/grid-filtering-template-sample.component.ts [hunk 2]
```diff
@@ -76,14 +76,12 @@ export class FilteringTemplateSampleComponent implements OnInit {
     }
 
     public onDateSelected(event, column: IgxColumnComponent) {
```
Comment (gedinakova) [inline src/app/grid/grid-filtering-template-sample/grid-filtering-template-sample.component.ts:78]: This code needs to be changed as follows in order for the clear button to work and actually clear the filter:
```
if (!event) {
            this.clearInput(column);
            return;
        }

        this._filterValues.set(column, event);
        this.grid1.filter(column.field, event, IgxDateFilteringOperand.instance().condition('equals'),
                column.filteringIgnoreCase);
```

**Review by gedinakova — CHANGES_REQUESTED** [review]

File: src/app/hierarchical-grid/hierarchical-grid-filtering-template/hierarchical-grid-filtering-template.component.ts [hunk 3]
```diff
@@ -34,7 +34,9 @@ export class HGridFilteringTemplateSampleComponent implements OnInit {
         };
     }
 
-    public formatter = (a) => a;
+    public formatter(date: Date): string {
```
Comment (gedinakova) [inline src/app/hierarchical-grid/hierarchical-grid-filtering-template/hierarchical-grid-filtering-template.component.ts:37]: This change is wrong - this formatter is used in columns like Debut, Launch Date, etc. They are supposed to show the year only. Currently, the formatter doesn't handle well the underlying value and always shows 1/1/1970.

**Review by gedinakova — CHANGES_REQUESTED** [review]

File: src/app/grid/grid-filtering-template-sample/grid-filtering-template-sample.component.html [hunk 4]
```diff
@@ -42,31 +42,14 @@
...
-                            [value]="displayData | date"
-                            [readonly]="true"/>
-                        <igx-suffix *ngIf="input.value || input.value === 0" (click)="clearInput(column)" tabindex="0">
-                            <igx-icon>clear</igx-icon>
-                        </igx-suffix>
-                    </igx-input-group>
-                </ng-template>
+            <igx-date-picker #dp [value]="getFilterValue(column)" (valueChange)="onDateSelected($event, column)" mode="dialog"
+                [displayDensity]="'compact'" placeholder="Filter..." type="box">
+                <igx-picker-toggle igxPrefix (click)="dp.open()">
```
Comment (wnvko) [inline src/app/grid/grid-filtering-template-sample/grid-filtering-template-sample.component.html:47]: do not handle `click` in the `igx-picker-toggle` - this should be handled by date picker

File: src/app/grid/grid-filtering-template-sample/grid-filtering-template-sample.component.html [hunk 5]
```diff
@@ -42,31 +42,14 @@
...
+                    <igx-icon>search</igx-icon>
+                </igx-picker-toggle>
+                <igx-picker-clear *ngIf="value" igxSuffix (click)="dp.clear()">
```
Comment (wnvko) [inline src/app/grid/grid-filtering-template-sample/grid-filtering-template-sample.component.html:50]: do not handle `click` in the `igx-picker-clear` - this should be handled by date picker

File: src/app/hierarchical-grid/hierarchical-grid-filtering-template/hierarchical-grid-filtering-template.component.html [hunk 6]
```diff
@@ -57,32 +57,14 @@
...
-                        [value]="displayData | date"
-                        [readonly]="true"/>
-                    <igx-suffix *ngIf="input.value || input.value === 0" (click)="clearInput(column, grid)" tabindex="0">
-                        <igx-icon>clear</igx-icon>
-                    </igx-suffix>
-                </igx-input-group>
-            </ng-template>
+        <igx-date-picker #dp [value]="getFilterValue(column)" (valueChange)="onDateSelected($event, column, grid)" mode="dialog"
+            [displayDensity]="'compact'" placeholder="Filter..." type="box">
+            <igx-picker-toggle igxPrefix (click)="dp.open()">
```
Comment (wnvko) [inline src/app/hierarchical-grid/hierarchical-grid-filtering-template/hierarchical-grid-filtering-template.component.html:62]: do not handle `click` in the `igx-picker-toggle` - this should be handled by date picker

File: src/app/hierarchical-grid/hierarchical-grid-filtering-template/hierarchical-grid-filtering-template.component.html [hunk 7]
```diff
@@ -57,32 +57,14 @@
...
+                <igx-icon>search</igx-icon>
+            </igx-picker-toggle>
+            <igx-picker-clear *ngIf="value" igxSuffix (click)="dp.clear()">
```
Comment (wnvko) [inline src/app/hierarchical-grid/hierarchical-grid-filtering-template/hierarchical-grid-filtering-template.component.html:65]: do not handle `click` in the `igx-picker-clear` - this should be handled by date picker

File: src/app/tree-grid/tree-grid-filtering-template-sample/tree-grid-filtering-template-sample.component.html [hunk 8]
```diff
@@ -43,31 +43,14 @@
...
-                                [value]="displayData | date"
-                                [readonly]="true" />
-                            <igx-suffix *ngIf="input.value || input.value === 0" (click)="clearInput(column)" tabindex="0">
-                                <igx-icon>clear</igx-icon>
-                            </igx-suffix>
-                        </igx-input-group>
-                    </ng-template>
+                <igx-date-picker #dp [value]="getFilterValue(column)" (valueChange)="onDateSelected($event, column)" mode="dialog"
+                    [displayDensity]="'compact'" placeholder="Filter..." type="box">
+                    <igx-picker-toggle igxPrefix (click)="dp.open()">
```
Comment (wnvko) [inline src/app/tree-grid/tree-grid-filtering-template-sample/tree-grid-filtering-template-sample.component.html:48]: do not handle `click` in the `igx-picker-toggle` - this should be handled by date picker

File: src/app/tree-grid/tree-grid-filtering-template-sample/tree-grid-filtering-template-sample.component.html [hunk 9]
```diff
@@ -43,31 +43,14 @@
...
+                        <igx-icon>search</igx-icon>
+                    </igx-picker-toggle>
+                    <igx-picker-clear *ngIf="value" igxSuffix (click)="dp.clear()">
```
Comment (wnvko) [inline src/app/tree-grid/tree-grid-filtering-template-sample/tree-grid-filtering-template-sample.component.html:51]: do not handle `click` in the `igx-picker-clear` - this should be handled by date picker

**Review by wnvko — CHANGES_REQUESTED** [review]

**Review by wnvko — DISMISSED** [review]

File: src/app/grid/grid-filtering-template-sample/grid-filtering-template-sample.component.ts [hunk 10]
```diff
@@ -20,6 +20,7 @@ export class FilteringTemplateSampleComponent implements OnInit {
     public grid1: IgxGridComponent;
 
     public data: any[];
+    public displayDateFormat = 'MMM d, y';
```
Comment (gedinakova) [inline src/app/grid/grid-filtering-template-sample/grid-filtering-template-sample.component.ts:23]: Let's update this format so it matches the data cells' format - 'M/d/y'.

File: src/app/tree-grid/tree-grid-filtering-template-sample/tree-grid-filtering-template-sample.component.ts [hunk 11]
```diff
@@ -20,6 +20,7 @@ export class TreeGridFilteringTemplateSampleComponent implements OnInit {
     public treegrid1: IgxTreeGridComponent;
 
     public data: any[];
+    public displayDateFormat = 'MMM d, y';
```
Comment (gedinakova) [inline src/app/tree-grid/tree-grid-filtering-template-sample/tree-grid-filtering-template-sample.component.ts:23]: Let's update this format so it matches the data cells' format - 'M/d/y'.

**Review by gedinakova — CHANGES_REQUESTED** [review]

**Review by gedinakova — APPROVED** [review]

**Pull request merged**